image/bmp.ksy

```diff
@@ -17,31 +17,104 @@ meta:
+
+  ## BITMAPV2INFOHEADER and BITMAPV3INFOHEADER
+
+  A beneficial discussion on Adobe forum (archived):
+  [**Invalid BMP Format with Alpha channel**](
+    https://web.archive.org/web/20150127132443/https://forums.adobe.com/message/3272950
+  )
+
+  In 2010, someone noticed that Photoshop generated BMP with an odd type of header. There wasn't
+  any documentation available for this header at the time (and still isn't).
+  However, Chris Cox (former Adobe employee) claimed that they hadn't invented any type
+  of proprietary header and everything they were writing was taken directly
+  from the Microsoft documentation.
+
+  It showed up that the unknown header was called BITMAPV3INFOHEADER.
+  Although Microsoft has apparently requested and verified the use of the header,
+  the documentation on MSDN has probably got lost and they have probably
+  forgotten about this type of header.
+
+  This is the only source I could find about these structures, so we could't rely
+  on it so much, but I think supporting them as a read-only format won't harm anything.
+  Due to the fact that it isn't documented anywhere else, most applications don't support it.
+
+  All Windows headers at once (including mentioned BITMAPV2INFOHEADER and BITMAPV3INFOHEADER):
+  ![Bitmap headers overview](
+    https://forums.adobe.com/servlet/JiveServlet/showImage/2-3273299-47801/BMP_Headers.png
+  )
+
+  ## Specs
+   * [Bitmap Storage (Windows Dev Center)](
+       https://docs.microsoft.com/en-us/windows/win32/gdi/bitmap-storage
+     )
+      * BITMAPFILEHEADER
+      * BITMAPINFOHEADER
+      * BITMAPV4HEADER
+      * BITMAPV5HEADER
+   * [OS/2 Bitmap File Format](
+        https://www.fileformat.info/format/os2bmp/egff.htm
+     )
+      * BITMAPFILEHEADER (OS2BMPFILEHEADER)
+      * BITMAPCOREHEADER (OS21XBITMAPHEADER)
+      * OS22XBITMAPHEADER
+   * [Microsoft Windows Bitmap](
+        http://netghost.narod.ru/gff/graphics/summary/micbmp.htm
+     )
+      * BITMAPFILEHEADER (WINBMPFILEHEADER)
+      * BITMAPCOREHEADER (WIN2XBITMAPHEADER)
+      * BITMAPINFOHEADER (WINNTBITMAPHEADER)
+      * BITMAPV4HEADER (WIN4XBITMAPHEADER)
+
 seq:
-  - id: file_hdr
+  - id: file_header
     type: file_header
-  - id: len_dib_header
-    type: s4
-  - id: dib_header
-    size: len_dib_header - 4
-    type:
-      switch-on: len_dib_header
-      cases:
-        12: bitmap_core_header
-        40: bitmap_info_header
-        104: bitmap_core_header # FIXME, should use BITMAPV4HEADER
-        124: bitmap_core_header # FIXME, should use BITMAPV5HEADER
+  - id: dib_info
+    size: file_header.ofs_bitmap - 14 # 14 = file_header._sizeof (TODO: replace when KSC 0.9 is released)
+    type: bitmap_info
+  - id: bitmap
+    type: bitmap
+    size-eos: true
 types:
+  bitmap:
+    doc: |
+      Replace with an opaque type if you care about the pixels.
+      You can look at an example of a JavaScript implementation: https://github.com/generalmimon/bmptool/blob/master/src/Bitmap.js
```

Comment:
Might want to add a link to https://github.com/kaitai-io/kaitai_struct/issues/188 with a lengthy discussion about exactly what you're talking about here.